hathor/p2p/states/peer_id.py

```diff
@@ -68,10 +74,16 @@ def send_peer_id(self) -> None:
         """
         protocol = self.protocol
         my_peer = protocol.my_peer
+
+        if not self.should_relay_ipv6_entrypoints:
+            entrypoints_as_str = my_peer.ipv4_entrypoints_as_str()
+        else:
+            entrypoints_as_str = my_peer.entrypoints_as_str()
```

Comment:
Should we restrict `my_peer.entrypoints_as_str()` if IPv4 is disabled? It could make sense to at least issue warnings if there are IPv4 addresses in `my_peer.entrypoints`.

Reply:
I did this way for simplicity. My objective with this check was to avoid relaying IPv6 to nodes that aren't ready to parse them.

All nodes are ready to parse IPv4, so it didn't seem necessary to limit them.

However, nodes that have IPv4 disabled will not try to connect to IPv4 entrypoints they receive.

But, thinking about your suggestion, isn't it better to let the IPv4 addresses be relayed even by nodes that have it disabled, so it could pass them forward to others that might use them? In some very specific (and unlikely) cases this could be useful, like if some node needs to first connect to some other that is IPv6-only as a bootstrap.

Reply:
I think it makes sense to try to propagate all IPs as much as possible (and just avoid sending IPv6 addresses to peers that can't parse them).